kafka_consumer/datadog_checks/kafka_consumer/kafka_consumer.py

```diff
@@ -16,12 +27,448 @@ class KafkaCheck(AgentCheck):
+                find_coordinator_future = self._kafka_client._find_coordinator_id_send_request(consumer_group)
+                find_coordinator_future.add_callback(self._find_coordinator_callback, consumer_group)
+                self._consumer_futures.append(find_coordinator_future)
+        else:
+            raise ConfigurationError(
+                "Cannot fetch consumer offsets because no consumer_groups are specified and "
+                "monitor_unlisted_consumer_groups is %s." % self.instance._monitor_unlisted_consumer_groups
+            )
+
+        # Loop until all futures resolved.
+        self._kafka_client._wait_for_futures(self._consumer_futures)
+        del self._consumer_futures  # since it's reset on every check run, no sense holding the reference between runs
+
+    def _list_groups_callback(self, broker_id, response):
+        """Callback that takes a ListGroupsResponse and issues an OffsetFetchRequest for each group.
+
+        broker_id must be manually passed in because it is not present in the response. Keeping track of the broker that
+        gave us this response lets us skip issuing FindCoordinatorRequests because Kafka brokers only include
+        consumer groups in their ListGroupsResponse when they are the coordinator for that group.
+        """
+        for consumer_group, group_type in self._kafka_client._list_consumer_groups_process_response(response):
+            # consumer groups from Kafka < 0.9 that store their offset in Kafka don't use Kafka for group-coordination
+            # so their group_type is empty
+            if group_type in ('consumer', ''):
+                single_group_offsets_future = self._kafka_client._list_consumer_group_offsets_send_request(
+                    group_id=consumer_group, group_coordinator_id=broker_id
+                )
+                single_group_offsets_future.add_callback(self._single_group_offsets_callback, consumer_group)
+                self._consumer_futures.append(single_group_offsets_future)
+
+    def _find_coordinator_callback(self, consumer_group, response):
+        """Callback that takes a FindCoordinatorResponse and issues an OffsetFetchRequest for the group.
+
+        consumer_group must be manually passed in because it is not present in the response, but we need it in order to
+        associate these offsets to the proper consumer group.
+
+        The OffsetFetchRequest is scoped to the topics and partitions that are specified in the check config. If
+        topics are unspecified, it will fetch all known offsets for that consumer group. Similiarly, if the partitions
+        are unspecified for a topic listed in the config, offsets are fetched for all the partitions within that topic.
+        """
+        coordinator_id = self._kafka_client._find_coordinator_id_process_response(response)
+        topics = self._consumer_groups[consumer_group]
+        if not topics:  # TODO will topics be None or empty list/set if unspecified??
```

Comment:
`None`

![Capture](https://user-images.githubusercontent.com/9677399/63393348-9a1bea00-c388-11e9-9b8c-0a34dd170dd0.PNG)

Reply:
Thank you!